ingestion_server/ingestion_server/tasks.py

```diff
@@ -2,155 +2,199 @@
+from enum import Enum, auto
+from multiprocessing import Process, Value
+from typing import Optional
 
 from ingestion_server import slack
+from ingestion_server.constants.media_types import MediaType
 from ingestion_server.indexer import TableIndexer, elasticsearch_connect
 from ingestion_server.ingest import reload_upstream
 
 
 class TaskTypes(Enum):
-    # Completely reindex all data for a given model.
-    REINDEX = 0
-    # Reindex updates to a model from the database since a certain date.
-    UPDATE_INDEX = 1
-    # Download the latest copy of the data from the upstream database, then
-    # completely reindex the newly imported data.
-    INGEST_UPSTREAM = 2
-    # Create indices in Elasticsearch for QA tests.
-    # This is not intended for production use, but can be safely executed in a
-    # production environment without consequence.
-    LOAD_TEST_DATA = 3
+    @staticmethod
+    def _generate_next_value_(name: str, *args, **kwargs) -> str:
+        """
+        Generates the value for ``auto()`` given the name of the enum item. Therefore,
+        this function must be defined before any of the enum items.
+
+        :param name: the enum variable name
+        :return: the enum value
+        """
+
+        return name.lower()
+
+    REINDEX = auto()
+    """completely reindex all data for a given model"""
+
+    INGEST_UPSTREAM = auto()
+    """download the latest copy of the data from the upstream database, then completely
+    reindex the newly imported data"""
+
+    UPDATE_INDEX = auto()  # TODO: delete eventually, rarely used
+    """reindex updates to a model from the database since the given date"""
+
+    POINT_ALIAS = auto()
+    """map a given index to a given alias, used when going live with an index"""
+
+    LOAD_TEST_DATA = auto()
+    """create indices in ES for QA tests; this is not intended to run in production but
+    can be run without negative consequences"""
+
+    def __str__(self):
+        """
+        Get the string representation of this enum. Unlike other objects, this
+        does not default to ``__repr__``.
+
+        :return: the string representation
+        """
+
+        return self.name
 
 
 class TaskTracker:
     def __init__(self):
-        self.id_task = {}
-        self.id_action = {}
-        self.id_progress = {}
-        self.id_start_time = {}
-        self.id_finish_time = {}
-        self.id_active_workers = {}
-
-    def add_task(self, task, task_id, action, progress, finish_time, active_workers):
-        self._prune_old_tasks()
-        self.id_task[task_id] = task
-        self.id_action[task_id] = action
-        self.id_progress[task_id] = progress
-        self.id_start_time[task_id] = dt.datetime.utcnow().timestamp()
-        self.id_finish_time[task_id] = finish_time
-        self.id_active_workers[task_id] = active_workers
-        return task_id
+        self.tasks = {}
 
     def _prune_old_tasks(self):
+        # TODO: Populate, document or delete function stub
```

Comment:
What is the purpose of this function now?

Reply:
To be honest, we could delete it (it doesn't do anything) but I wanted to be sure that there was nothing planned for it before we removed it from the codebase.